compiler/rustc_mir/src/util/pretty.rs

```diff
@@ -273,7 +273,12 @@ pub fn write_mir_pretty<'tcx>(
 
     let mut first = true;
     for def_id in dump_mir_def_ids(tcx, single) {
-        let body = &tcx.optimized_mir(def_id);
+        let body = match tcx.hir().body_const_context(def_id.expect_local()) {
+            // For `const fn` we want to render the optimized MIR. If you want the mir used in
+            // ctfe, you can dump the MIR after the `Deaggregator` optimization pass.
+            None | Some(rustc_hir::ConstContext::ConstFn) => tcx.optimized_mir(def_id),
```

Comment:
@RalfJung should we print both the `mir_for_ctfe` and the `optimized_mir` for `const fn` (so it would show up on the playground dump), or is it enough that we have an unstable API for dumping the MIR after each mir pass?

Reply:
Good questions. I am not a big fan of the 2nd idea since getting the right MIR out of that can be tricky. But another option might be to have an (unstable?) `--emit const-mir` flag.

Reply:
I implemented it in a way that prints both optimized and unoptimized MIR.